addons/project/models/project_task_recurrence.py

```diff
@@ -79,11 +79,28 @@ def _get_recurrence_delta(self):
             f"{self.repeat_unit}s": self.repeat_interval
         })
 
+    def _copy_task_dependency(self, task_copy, task_original):
+        if 'task_mapping' not in self.env.context:
+            self = self.with_context(task_mapping=dict())
+
+        for i in range(len(task_original.child_ids)):
+            self._copy_task_dependency(task_copy.child_ids[i], task_original.child_ids[i])
+
+        task_mapping = self.env.context.get('task_mapping')
+        task_mapping[task_original.id] = task_copy.id
+        tasks_copied = task_mapping.values()
+        task_original.write({'depend_on_ids': [Command.unlink(t.id) for t in task_original.depend_on_ids if t.id in tasks_copied]})
+        task_original.write({'dependent_ids': [Command.unlink(t.id) for t in task_original.dependent_ids if t.id in tasks_copied]})
+        task_copy.write({'depend_on_ids': [Command.link(task_mapping.get(t.id, t.id)) for t in task_original.depend_on_ids]})
+        task_copy.write({'dependent_ids': [Command.link(task_mapping.get(t.id, t.id)) for t in task_original.dependent_ids]})
```

Comment:
There are 2 things that i'm not found of here.
1) using write is usually not the best practice. it's better to simply use task.copy.depends_on = new_value instead. That way, we let the orm optimize the data updates. Unless we have a reason to specifically use 'write' in this case.
2) the fact that you have to 'unlink' some dependence on the original task, means that at some point you're linking things that are not supposed to be linked. We should try to simply not make those useless link in the first place, instead of trying to fix our 'mistake' later on.